core/shared/src/main/scala/scalaz/zio/Schedule.scala

```diff
@@ -32,25 +32,25 @@ trait Schedule[-A, +B] extends Serializable { self =>
   /**
    * The initial state of the schedule.
    */
-  val initial: IO[Nothing, State]
+  val initial: Clock => IO[Nothing, State]
 
   /**
    * Updates the schedule based on a new input and the current state.
    */
-  val update: (A, State) => IO[Nothing, Schedule.Decision[State, B]]
+  val update: (A, State, Clock) => IO[Nothing, Schedule.Decision[State, B]]
 
   /**
    * Runs the schedule on the provided list of inputs, returning a list of
    * durations and outputs. This method is useful for testing complicated
    * schedules. Only as many inputs will be used as necessary to run the
    * schedule to completion, and additional inputs will be discarded.
    */
-  def run(as: Iterable[A]): IO[Nothing, List[(Duration, B)]] = {
+  def run(as: Iterable[A], clock: Clock = Clock.Live): IO[Nothing, List[(Duration, B)]] = {
```

Comment:
I am a fan of making this `Clock` not have a default. The reason is that it is designed for testing and a user should probably be forced to provide either their own clock, or an (upcoming) `Clock.Test`.

Reply:
Hi John, that's fine by me :-) however the [original issue](https://github.com/scalaz/scalaz-zio/issues/329) said that this would be nice as a default - so that's why I put it in

Reply:
...anyway it's now solved by [that same commit](https://github.com/scalaz/scalaz-zio/pull/332/commits/5fa247a94df8893f299bc2a415484e90bf9ce81a)